src/V3Ast.cpp

```diff
@@ -1141,6 +1141,96 @@ void AstNode::dumpTreeFile(const string& filename, bool append, bool doDump, boo
     }
 }
 
+static std::string nodeAddr(const AstNode* nodep) {
+        return v3Global.opt.dumpTreeAddrids() ? v3Global.ptrToId(nodep) : cvtToHex(nodep);
+}
+
+void AstNode::dumpTreeDotNode(std::ostream& str) const {
+    str << "\tn" << nodeAddr(this) << "\t[fontsize=8 "
+        << "label=\"" << typeName() << "\\n" <<name()
+        << "\""
+        << ", color=" << "black"
+        << ", shape=" << "ellipse"
+        << ", style=" << "bold"
+        << "];\n";
+    if (user1p()) {
+        str << "\tn" << nodeAddr(this) << " -> "<< nodeAddr(user1p())
+            << " ["
+            << "fontsize=8 label=\"" << "u1" << "\""
+            << " weight=" << 1 << " color=" << "red";
+        str << "];\n";
+    }
```

Comment:
Suggest make a little function that takes in user1p(), and "1", and prints to avoid duplication below.
Also perhaps use different color than red to disambiguate from other edges.

Reply:
Also, "weight" and "fontsize" (possibly others as well) seems to be constant  everywhere, so it would be best set these globally (or omit) as not to fill up the disk. Although this format is most likely only helpful for smallish inputs (I am not aware of a dot viewer that handles very large graphs well)

Reply:
BTW, `userNp` is quite often not a pointer to an AstNode (but a flag/integer, or a pointer to some other data structure), so I don't think this will be very clear as is. 

One could go and gather all the node addresses on a first pass and then dump these  userp edges only if it holds an actual node address.

Reply:
I thought about this too. I think the existing code will work, the edge to a non-node will make a vertex that just looks different from a tree node.  Worth testing.

Reply:
Yeah, also TBH it's very rare that userps persist across passes, so these will likely be all nullptr where we dump trees in the standard way.